pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -259,6 +260,18 @@ private DataTable processQuery(List<IndexSegment> indexSegments, QueryContext qu
       ExecutorService executorService, @Nullable StreamObserver<Server.ServerResponse> responseObserver, long endTimeMs,
       boolean enableStreaming)
       throws Exception {
+
+    // Validate whether column names in the query are valid
+    Set<String> columnNamesFromSchema = _instanceDataManager.getColumnNamesByTable(queryContext.getTableName());
+    Set<String> columnNamesFromQuery = queryContext.getColumns();
+    if (!columnNamesFromSchema.isEmpty() && !columnNamesFromSchema.containsAll(columnNamesFromQuery)) {
+      columnNamesFromQuery.removeAll(columnNamesFromSchema);
```

Comment:
Using `containsAll()` and `removeAll` seems inefficient.

Reply:
The `containsAll` method has already been used in `DataSchemaSegmentPruner` class as well. And it gets called for every selected segment. And here it is just done once. 
The method `removeAll` here is invoked only when the if statement is satisfied, so no extra execution on querying actual segments.

Reply:
How about aliasing? Will it work for `select foo as bar from table`?

Reply:
Aliasing has already been taken care of when generating the queryContext in pinot-server. So any queries with alias will 
 still work here.

Reply:
Ok, great. Please add tests to cover aliasing case (if they don't already exist).

Reply:
Aliasing case has been covered in integration tests. We should be good here. :)